sonic-net/sonic-buildimage
# PDDF: Supporting extra system fans in the common PDDF drivers

#### Why I did it
Current PDDF design supports only 12 individual fans (if 2 fans per tray then total of 6 fantrays). However, some platform have more fans. To support those platforms via PDDF, we added support for more fans in common fan PDDF drivers.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
I did it by adding extra 4 fans in the pddf_fan_driver. This includes adding extra sysfs for all the supported attributes for the newly added fans.

#### How to verify it

1. Check there are no errors while loading the pddf_fan_driver.ko
2. Check 'show platform fan' command output.

Tested the changes on AS7816 (TH2) platform. Here are the results.

```
root@sonic:/home/admin# show ver |head -10

SONiC Software Version: SONiC.master.0-dirty-20230718.022417
SONiC OS Version: 11
Distribution: Debian 11.7
Kernel: 5.10.0-18-2-amd64
Build commit: 0cdac42ea
Build date: Tue Jul 18 09:57:17 UTC 2023
Built by: fk410167@sonic-lvn-csg-002

Platform: x86_64-accton_as7816_64x-r0
root@sonic:/home/admin#
root@sonic:/home/admin#
root@sonic:/home/admin# show platform summary
Platform: x86_64-accton_as7816_64x-r0
HwSKU: Accton-AS7816-64X
ASIC: broadcom
ASIC Count: 1
Serial Number: AAA1903AAEV
Model Number: FP3AT7664000A
Hardware Revision: N/A
root@sonic:/home/admin# show log pddf_fan_driver
Jul 13 10:53:56 sonic pddf_util.py: [PDDF]: 'modprobe pddf_fan_driver_module' with result: 0    output:
Jul 13 10:53:56 sonic pddf_util.py: [PDDF]: 'modprobe pddf_fan_driver_module' with result: 0    output:
root@sonic:/home/admin#
root@sonic:/home/admin# show platform fan
  Drawer    LED         FAN    Speed    Direction    Presence    Status          Timestamp
--------  -----  ----------  -------  -----------  ----------  --------  -----------------
Fantray1  green  Fantray1_1      52%      EXHAUST     Present        OK  20230725 06:32:10
Fantray1  green  Fantray1_2      52%      EXHAUST     Present        OK  20230725 06:32:10
Fantray2  green  Fantray2_1      52%      EXHAUST     Present        OK  20230725 06:32:10
Fantray2  green  Fantray2_2      52%      EXHAUST     Present        OK  20230725 06:32:10
Fantray3  green  Fantray3_1      52%      EXHAUST     Present        OK  20230725 06:32:10
Fantray3  green  Fantray3_2      52%      EXHAUST     Present        OK  20230725 06:32:10
Fantray4  green  Fantray4_1      52%      EXHAUST     Present        OK  20230725 06:32:10
Fantray4  green  Fantray4_2      52%      EXHAUST     Present        OK  20230725 06:32:10
     N/A    N/A   PSU1_FAN1       0%      exhaust     Present    Not OK  20230725 06:32:11
     N/A    N/A   PSU2_FAN1      78%      exhaust     Present        OK  20230725 06:32:11
root@sonic:/home/admin#
root@sonic:/home/admin# show platform temperature
                   Sensor    Temperature    High TH    Low TH    Crit High TH    Crit Low TH    Warning          Timestamp
-------------------------  -------------  ---------  --------  --------------  -------------  ---------  -----------------
      CpuBoard_temp(0x4B)           24.5       80.0       N/A             N/A            N/A      False  20230725 06:32:11
          FB_1_temp(0x4D)           26.5       80.0       N/A             N/A            N/A      False  20230725 06:32:11
          FB_2_temp(0x4E)           26.5       80.0       N/A             N/A            N/A      False  20230725 06:32:11
        MB_MAC_temp(0x48)           30         80.0       N/A             N/A            N/A      False  20230725 06:32:11
 MB_RearCenter_temp(0x49)           33.5       80.0       N/A             N/A            N/A      False  20230725 06:32:11
MB_RightCenter_temp(0x4A)           24.5       80.0       N/A             N/A            N/A      False  20230725 06:32:11
               PSU1_TEMP1            0          N/A       N/A             N/A            N/A      False  20230725 06:32:11
               PSU2_TEMP1           21          N/A       N/A             N/A            N/A      False  20230725 06:32:11
root@sonic:/home/admin# 

```



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
